crates/proof-of-sql/src/base/database/column_struct_operation.rs

```diff
@@ -0,0 +1,2284 @@
+            lhs_data
+                .iter()
+                .zip(rhs_data.iter())
+                .map(|(lhs, rhs)| op(lhs * scaling_factor, *rhs))
+                .collect()
+        }
+        Ordering::Greater => {
+            let scaling_factor = 10i64.pow((lhs_scale - rhs_scale) as u32);
+            lhs_data
+                .iter()
+                .zip(rhs_data.iter())
+                .map(|(lhs, rhs)| op(*lhs, rhs * scaling_factor))
+                .collect()
+        }
+        Ordering::Equal => lhs_data
+            .iter()
+            .zip(rhs_data.iter())
+            .map(|(lhs, rhs)| op(*lhs, *rhs))
+            .collect(),
+    }
+}
+
+impl<'a, S: Scalar> Column<'a, S> {
+    /// Element-wise NOT operation for a column
+    pub fn element_wise_not(&self, alloc: &'a Bump) -> ColumnOperationResult<Self> {
+        match self {
+            Self::Boolean(values) => Ok(Self::Boolean(
+                alloc.alloc_slice_copy(slice_not(values).as_ref()),
+            )),
+            _ => Err(ColumnOperationError::UnaryOperationInvalidColumnType {
+                operator: UnaryOperator::Not,
+                operand_type: self.column_type(),
+            }),
+        }
+    }
+
+    /// Element-wise AND for two columns
+    pub fn element_wise_and(&self, rhs: &Self, alloc: &'a Bump) -> ColumnOperationResult<Self> {
+        if self.len() != rhs.len() {
+            return Err(ColumnOperationError::DifferentColumnLength {
+                len_a: self.len(),
+                len_b: rhs.len(),
+            });
+        }
+        match (self, rhs) {
+            (Self::Boolean(lhs), Self::Boolean(rhs)) => Ok(Self::Boolean(
+                alloc.alloc_slice_copy(slice_and(lhs, rhs).as_ref()),
+            )),
+            _ => Err(ColumnOperationError::BinaryOperationInvalidColumnType {
+                operator: BinaryOperator::And,
+                left_type: self.column_type(),
+                right_type: rhs.column_type(),
+            }),
+        }
+    }
+
+    /// Element-wise OR for two columns
+    pub fn element_wise_or(&self, rhs: &Self, alloc: &'a Bump) -> ColumnOperationResult<Self> {
+        if self.len() != rhs.len() {
+            return Err(ColumnOperationError::DifferentColumnLength {
+                len_a: self.len(),
+                len_b: rhs.len(),
+            });
+        }
+        match (self, rhs) {
+            (Self::Boolean(lhs), Self::Boolean(rhs)) => Ok(Self::Boolean(
+                alloc.alloc_slice_copy(slice_or(lhs, rhs).as_ref()),
+            )),
+            _ => Err(ColumnOperationError::BinaryOperationInvalidColumnType {
+                operator: BinaryOperator::Or,
+                left_type: self.column_type(),
+                right_type: rhs.column_type(),
+            }),
+        }
+    }
+
+    /// Element-wise equality check for two columns
+    #[allow(clippy::too_many_lines)]
+    pub fn element_wise_eq(&self, rhs: &Self, alloc: &'a Bump) -> ColumnOperationResult<Self> {
```

Comment:
Since all of these functions are identical (I think), could we just replace them all with a single function that we can pass in a few slice_op parameters?

```suggestion
    pub fn element_wise_binary_compare_op<F,G,H>(&self, rhs: &Self, alloc: &'a Bump, slice_op: F, slice_op_with_casting: G, decimal_slice_op: H) -> ColumnOperationResult<Self> {
```

Not sure if the generic parameters are needed.